testsuite/classlibrary/TestPprotect.sc

```diff
@@ -19,4 +20,75 @@ TestPprotect : UnitTest {
 		cond.hang;
 		this.assert(success, "Pprotect should clear the stream's exceptionHandler");
 	}
+
+	test_stream_can_be_restarted_after_error {
+		var pat, stream, hasRun = false;
+		pat = Pprotect(
+			Prout {
+				0.01.yield;
+				hasRun = true;
+				Error("dummy error").throw
+			},
+			{ stream.streamError }
+		);
+		stream = pat.play;
+		0.02.wait;
```

Comment:
is there no better way to do this?

Reply:
I don't see any. We are testing the play behavior, so we have to do something like this.

Reply:
no callback or signal or anything?

Reply:
I just noticed the first test of `TestPprotect` does use a condition to avoid this. I tried, but couldn't make use of this technique in the other cases.

Reply:
This form works:

```
	test_stream_can_be_restarted_after_error {
		var pat, stream, hasRun = false, condition = Condition.new;
		pat = Pprotect(
			Prout {
				0.yield;
				hasRun = true;
				Error("dummy error").throw
			},
			{ stream.streamError; condition.unhang }
		);
		stream = pat.play;
		condition.hang;
		hasRun = false;
		stream.reset;
		stream.play;
		condition.hang;
		this.assert(hasRun, "stream should be resettable after an error");
	}
```

Reply:
Well, I should say, the above form behaves correctly if the test passes. On failure, there's a very high likelihood of the test suite pausing indefinitely, with no report of which is the offending test. So we still need new features in the test framework to allow the above types of tests to fail with a meaningful report.

Reply:
I've added a fallback for this. We should write an interface for UnitTest that encapsulates (something like) this.